drivers/ethernet/eth_stm32_hal.c

```diff
@@ -306,6 +354,69 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 	}
 #endif /* CONFIG_SOC_SERIES_STM32H7X */
 
+#if defined(CONFIG_PTP_CLOCK_STM32_HAL)
+	if (timestamped_frame) {
+		/* Retrieve transmission timestamp from last DMA TX descriptor */
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		ETH_TxDescListTypeDef * dma_tx_desc_list;
+
+		__IO ETH_DMADescTypeDef *last_dma_tx_desc;
+
+		dma_tx_desc_list = &heth->TxDescList;
+		for (uint32_t i = 0; i < ETH_TX_DESC_CNT; i++) {
+			uint32_t last_desc_idx;
+
+			last_desc_idx = (cur_tx_desc_idx + i) % ETH_TX_DESC_CNT;
```

Comment:
```suggestion
              uint32_t last_desc_idx;
		for (uint32_t i = 0; i < ETH_TX_DESC_CNT; i++) {
			last_desc_idx = (cur_tx_desc_idx + i) % ETH_TX_DESC_CNT;
```

... or use const uint32_t last_desc_idx = (cur_tx_desc_idx + i) % ETH_TX_DESC_CNT; in the cacle alternatively.

Reply:
Let me try const uint32_t option.

Reply:
I got dinged because of lack of space, so I had to add space after const uint32_t:
```
-:219: WARNING:LINE_SPACING: Missing a blank line after declarations
#219: FILE: drivers/ethernet/eth_stm32_hal.c:368:
+			const uint32_t last_desc_idx = (cur_tx_desc_idx + i) % ETH_TX_DESC_CNT;
+			last_dma_tx_desc =
```